### PR TITLE
Bug fixes ng

### LINKED ISSUE #139: User should see visual confirmation that park has been saved
This feature had already been implemented, but a bug introduced by integration with the back-end has led to this new issue.

### LINKED ISSUE #140: User should always see park ratings as decimal numbers

### LINKED ISSUE #142: User should not be able to submit empty email field

### PR DESCRIPTION
**Pull request by nicolegooden**
### Participating Group Members:

- @nicolegooden 

### Code Highlights:

- Fixes bug so that save button text changes to confirm that a park has been saved (both upon saving live and retrieving a previously saved park)
- Gives user the option to sort by park rating (highest to lowest) when viewing in a list
- Changes order of events when user submits email address for sign-in 
- Changes welcome message in `TheHome` from mention of play dates to a message that better reflects our MVP
- When retrieving saved parks from the API, parses the park ratings using `parseFloat( )` rather than `parseInt( )` so that saved parks don't have a consistent rating of a whole number 
- Disables submit button for email if input is empty

### Have Tests Been Added?

- [X] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?

Some of these changes are a reflection of my exploration of the app after it had been submitted.  I noticed the bug with the saved park ratings in `apiCalls` and thought some of our messages to the user could be more clear.  

### Message/Questions for reviewer:

Right now, if the user searches for parks, views them in a list, then clicks the `sort` button, the functionality is working smoothly; however, if the user navigates to a different component/view and then returns to the results list view later, the parks will show up sorted but the `sort` button will be there instead of the message.  The button can still be clicked and the message will show up again (as we want), but the results would have already been sorted and therefore there are no visible changes to the list.  

The cause of this is that this `sorted` data in `ResultsList` is only local and is not stored in Vuex.  I am not sure if this specific data is worth storing at that level.  `sort( )` mutates the data for `searchResults`, so the ordered list persists across view changes and time (until user closes the browser or refreshes).  

Any preferences / suggestions for how I handle this minor bug moving forward?

### Issues:

- Addresses #26 
- Closes #139 
- Closes #140 
- Closes #142 

### Screenshots (if appropriate):

Screen Shot 2021-01-13 at 7 20 24 PM
Screen Shot 2021-01-13 at 7 20 42 PM
Screen Shot 2021-01-13 at 7 21 53 PM

Before `Sort By Rating` is clicked:

Screen Shot 2021-01-13 at 7 50 57 PM

After:

Screen Shot 2021-01-13 at 7 51 14 PM

### Tracking Consistency:

- [X] added appropriate labels
- [X] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [X] All new and existing tests passed

- [X] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const getSaved = (email) => {
             open_now: (park.opening_hours === 'true')
           },
           photos: [{ photo_reference: park.photo }],
-          rating: parseInt(park.rating)
+          rating: parseFloat(park.rating)
```
Comment (npdarrington) [inline src/apiCalls.js:57]: Nice attention to detail for this.

File: src/views/TheHome.vue [hunk 2]
```diff
@@ -5,7 +5,7 @@
       alt="picture of two puppies in the snow"
       src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSrnUwJ9xC5z-HtpYQZc3jHYJfcMtQUf8qzVg&usqp=CAU"
     />
-    <h2>Find your pup a play date at the best dog parks around!</h2>
+    <h2>Find your pup a new park to bark all over</h2>
```
Comment (npdarrington) [inline src/views/TheHome.vue:8]: Very cute wording!

**Review by npdarrington — APPROVED** [review]
Thanks for looking into these bugs Nicole and giving us a head start for the code fair! This will make it much more complete for the presentation on Friday.

File: src/views/TheHome.vue (hunk 2 above)
Comment (nicolegooden) [reply src/views/TheHome.vue:8]: I really do try... haha!

Comment (stacyp2006) [reply src/views/TheHome.vue:8]: So fun!

File: src/components/Email.vue [hunk 3]
```diff
@@ -38,15 +38,15 @@ export default {
...
       this.$store.commit('saveParks', fetchedSavedParks)
-      this.message = `Success! ${this.$store.state.email} has been saved.`
-      setTimeout(() => this.$store.commit('changeToSearch'), 2500);
     }
   },
   computed: {
     isInvalid() {
-      return this.email === '' || !this.email.includes('@') || !this.email.includes('.')
+      return !this.email.includes('@') || !this.email.includes('.')
     }
```
Comment (stacyp2006) [inline src/components/Email.vue:50]: Nice. I like this refactor with the disabled class on the button.

**Review by stacyp2006 — APPROVED** [review]

File: src/components/ResultsItemDetails.vue [hunk 4]
```diff
@@ -46,11 +46,10 @@ export default {
       return this.park.opening_hours.open_now === true ? 'Open Now' : 'Closed Now'
     },
     saved() {
-      if (this.$store.state.savedParks.includes(this.park)) {
-        return 'PARK SAVED!'
-      } else {
-        return 'SAVE'
-      }
+      let result = this.$store.state.savedParks.find(saved => {
```
Comment (bretm9) [inline src/components/ResultsItemDetails.vue:49]: Thanks for fixing the save button with this logic!

File: src/components/ResultsList.vue [hunk 5]
```diff
@@ -4,9 +4,13 @@
       <h5>{{ message }}</h5>
       <FindPark v-if="noResults" />
     </h4>
-    <section v-else :key='i' v-for='(result, i) in searchResults'>
+    <section v-else>
+    <button v-if="!sorted" @click="sortByRating">Sort By Rating</button>
```
Comment (bretm9) [inline src/components/ResultsList.vue:8]: This is a nice additional feature and continues to strengthen our excellent UX

File: src/components/ResultsList.vue [hunk 6]
```diff
@@ -45,8 +56,22 @@ export default {
 }
 </script>
 
-<style scoped>
+<style lang="scss" scoped>
```
Comment (bretm9) [inline src/components/ResultsList.vue:59]: Thanks for refactoring this to SASS

**Review by bretm9 — APPROVED** [review]
Thanks for these improvements!

**Pull request merged**